File: API/PCPartPicker/Models/Memory.cs
﻿using PCPartPicker.Interfaces;
using System;
using System.Collections.Generic;

namespace PCPartPicker.Models;

public partial class Memory : IComponent
{
    public int Id { get; set; }

    public string? Name { get; set; }

    public decimal? Price { get; set; }

    public decimal? Speed { get; set; }

    public string? Modules { get; set; }

    public decimal? PricePerGb { get; set; }

    public string? Color { get; set; }

    public decimal? FirstWordLatency { get; set; }

    public decimal? CasLatency { get; set; }

    public int? ManufacturerTypeId { get; set; }

    public int? Type { get; set; }

    public virtual ICollection<Configuration> Configurations { get; set; } = new List<Configuration>();

    public virtual ManufacturerType? ManufacturerType { get; set; }

    public virtual MemoryType? TypeNavigation { get; set; }
}

File: UI/pc-part-picker/src/model/Memory.ts
import { ManufacturerType } from "./ManufacturerType";
import { MemoryType } from "./MemoryType";
import { PcPart } from "./Pcpart";

export interface Memory extends PcPart{
    speed: number | null;
    modules: string | null;
    pricePerGb: number | null;
    color: string | null;
    firstWordLatency: number | null;
    casLatency: number | null;
    manufacturerTypeId: number | null;
    type: number | null;
    manufacturerType: string | null;
    typeNavigation: string | null;
}